scripts/sign_windows_executable.js

```diff
@@ -0,0 +1,169 @@
+      break;
+    case 'darwin':
+      args.push('--keystore', 'scripts/digicert-usb-config/eToken-macos.cfg');
+      break;
+    default:
+      throw new Error(`we do not support ev signing on ${process.platform}`);
+  }
+}
+
+/**
+ * Sign the target exeFile using a specific algorithm and options.
+ * @param {string} exeFile the full path of the exe file to be signed.
+ * @param {'sha1'|'sha256'} algorithm the algorithm used for signing.
+ * @param {object} options the additional options (cli arguments) for signing.
+ */
+async function signWindowsExecutable(exeFile, algorithm, options) {
+  const type = getOptionValue(options, 'certtype',
+    'WINDOWS_SIGNING_CERT_TYPE', false);
+  if (!type || type == 'none') {
+    console.info(`skip signing "${exeFile}"`);
+    return;
+  }
+
+  assert(!!exeFile, 'executable path is required');
+  assert(algorithm == 'sha1' || algorithm == 'sha256',
+    'hashing algorithm must be either "sha1" or "sha256"');
+
+  exeFile = path.resolve(exeFile);
+  await assertFileExists(exeFile, 'executable file "%s" does not exist');
+
+  const password = getOptionValue(options, 'password',
+    'WINDOWS_SIGNING_CERT_PASSWORD', true);
+
+  var jsignArgs = [
+    '--alg', algorithm == 'sha256' ? 'SHA-256' : 'SHA-1',
+    '--tsaurl', 'http://timestamp.digicert.com',
+    '--storepass', password];
+
+  switch (type) {
+    case 'pfx':
+      appendPfxJsignArgs(jsignArgs, options);
+      break;
+    case 'usb':
+      appendDigicertUsbJsignArgs(jsignArgs, options);
+      break;
+    default:
+      throw new Error(`cert type ${type} is not supported`);
+  }
+
+  // This is how we use ES6 modules in CommonJS
+  const { default: jsign } = await import('./jsign.mjs');
+  const exitCode = await jsign(exeFile, jsignArgs);
+  if (exitCode == 0) {
+    console.info(`successfully signed "${exeFile}"`);
+  } else {
+    console.error(`jsign exited with error code ${exitCode}`);
+    throw new Error(`failed to sign "${exeFile}"`);
+  }
+}
+
+/**
+ * The entry point which will be called by electron-builder signing module.
+ * @param {Object} configuration a configuration containing signing information
+ * @param {string} configuration.path the executable file path to be signed
+ * @param {'sha1'|'sha256'} configuration.hash requested hash algorithm
+ * @param {boolean} configuration.isNest whether it is a secondary signature
+ * @param {Object} configuration.options a duplication of electron-builder.json
+ */
+async function electronBuilderEntryPoint(configuration) {
+  await signWindowsExecutable(configuration.path, configuration.hash, null);
+}
+
+/**
+ * The entry point when calling this script from cli.
+ */
+async function cliEntryPoint() {
+  const {target, algorithm, ...options} = minimist(process.argv);
+  await signWindowsExecutable(target, algorithm, options);
+}
+
+// CommonJS module is required, ES6 module is not supported by electron-builder
+exports.default = electronBuilderEntryPoint;
```

Comment:
Can you create a separate file for the electron-builder signing that depends on this library?? The multiple entry points is confusing.

Reply:
Sure thing. Then the imports would be much clearer as well.

Reply:
Updated in the new files (.cjs & .mjs).